src/Search/SimilarDocumentsSearchResult.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Meilisearch\Search;
+
+class SimilarDocumentsSearchResult implements \Countable, \IteratorAggregate
+{
+    /**
+     * @var array<int, array<string, mixed>>
+     */
+    private array $hits;
+
+    /**
+     * `estimatedTotalHits` is the attributes returned by the Meilisearch server
+     * and its value will not be modified by the methods in this class.
+     * Please, use `hitsCount` if you want to know the real size of the `hits` array at any time.
+     */
+    private int $estimatedTotalHits;
+    private int $hitsCount;
+    private int $offset;
+    private int $limit;
+    private int $processingTimeMs;
+
+    private string $id;
+
+    public function __construct(array $body)
+    {
+        $this->id = $body['id'];
+        $this->hits = $body['hits'] ?? [];
```

Comment:
Is there are case when hits key would not exist? Because the line below you don't use null-coalescing like here

Reply:
fixed